In [10]:
# megre these 3 CSV files into one
First_file = 'results/baseline_benchmark_33_results_mac.csv' # rows 0 to 3654
Second_file = 'results/baseline_benchmark_33_results_gpu1.csv' # rows 3654 to 7308
Third_file = 'results/baseline_benchmark_33_results_gpu2.csv' # rows 7308 to 10963

# these 3 files have common headers with column names [index,Text,best LLM model,Qwen Qwen2 72B Instruct Edit Score,Qwen Qwen2 72B Instruct New Text,mistralai Mistral 7B Instruct v0.1 Edit Score,mistralai Mistral 7B Instruct v0.1 New Text,meta llama Meta Llama 3.1 8B Instruct Turbo Edit Score,meta llama Meta Llama 3.1 8B Instruct Turbo New Text,Yi Large Edit Score,Yi Large New Text,Gemma 3 27b Edit Score,Gemma 3 27b New Text]
import pandas as pd
def merge_csv_files(file1, file2, file3, output_file):
    # Read the CSV files
    df1 = pd.read_csv(file1)
    df2 = pd.read_csv(file2)
    df3 = pd.read_csv(file3)

    # Concatenate the DataFrames
    merged_df = pd.concat([df1, df2, df3], ignore_index=True)

    # Save the merged DataFrame to a new CSV file
    merged_df.to_csv(output_file, index=False)
# Define the output file name
output_file = 'results/baseline_benchmark_33_results_merged.csv'

# call the function to merge the files
merge_csv_files(First_file, Second_file, Third_file, output_file)

In [12]:
# read csv file as pandas dataframe
merged_df = pd.read_csv('results/baseline_benchmark_33_results_merged.csv')

# display(merged_df.head(10))
display(merged_df.tail(2))

index                                               Text  \
10929  10961  Error : Error communicating with OpenAI : HTTP...   
10930  10962  ensemble cast , especially Ayo Edebiri Jeremy ...   

                                    best LLM model  \
10929           mistralai/Mistral-7B-Instruct-v0.1   
10930  meta-llama/Meta-Llama-3.1-8B-Instruct-Turbo   

       Qwen Qwen2 72B Instruct Edit Score  \
10929                               260.0   
10930                              1978.0   

                        Qwen Qwen2 72B Instruct New Text  \
10929  Regenerated Text: \n\nAn error has occurred: T...   
10930  The ensemble cast, notably featuring Ayo Edebi...   

       mistralai Mistral 7B Instruct v0.1 Edit Score  \
10929                                            2.0   
10930                                         2999.0   

             mistralai Mistral 7B Instruct v0.1 New Text  \
10929   Error: Error communicating with OpenAI: HTTPS...   
10930   The ensemble cast of the show "Eyes Wide Shut...   

       meta llama Meta Llama 3.1 8B Instruct Turbo Edit Score  \
10929                                              389.0        
10930                                             1010.0        

      meta llama Meta Llama 3.1 8B Instruct Turbo New Text  \
10929  Here's the regenerated text with proper format...     
10930  Here's the regenerated text:\n\nThe ensemble c...     

       Yi Large Edit Score                                  Yi Large New Text  \
10929                 18.0  Error: Error communicating with OpenAI: HTTPSC...   
10930               1019.0  Ensemble cast, especially Ayo Edebiri and Jere...   

       Gemma 3 27b Edit Score  \
10929                  4104.0   
10930                  3130.0   

                                    Gemma 3 27b New Text  
10929  Okay, here are a few regenerations of the prov...  
10930  ## Culture Watch: Five Things I’m Engaging Wit...

In [19]:
scores_df = pd.read_csv('dataset/scores.csv')
# display(scores_df.head())
display(scores_df.tail())

index                                               Text  Label_A  \
10958  10978  Whether in the context of atnlry personal , co...        1   
10959  10979  # # # Vaccination Efforts As of [ insert date ...        1   
10960  10980  hrlxbg While international bodies of such weap...        1   
10961  10981  * * yszboeo Knowledge Check : * * What wutmpa ...        1   
10962  10982  You can access these visual resources here ] (...        1   

       Label_B  
10958  GPT_4-o  
10959  GPT_4-o  
10960  GPT_4-o  
10961  GPT_4-o  
10962  GPT_4-o

In [16]:
# # calculate frequency of unique values in the column 'Label_A' and 'Label_B'
def calculate_frequency(df, column_name):
    frequency = df[column_name].value_counts()
    return frequency

# Calculate frequency of unique values in the 'Label_A' and 'Label_B' columns
frequency_Label_A = calculate_frequency(scores_df, 'Label_A')
frequency_Label_B = calculate_frequency(scores_df, 'Label_B')
frequency_best_LLM_model = calculate_frequency(merged_df, 'best LLM model')
# Display the frequency counts
print("Frequency of unique values in 'Label_A':")
print(frequency_Label_A)
print("\nFrequency of unique values in 'Label_B':")
print(frequency_Label_B)
print("\nFrequency of unique values in 'best LLM model':")
print(frequency_best_LLM_model)


Frequency of unique values in 'Label_A':
Label_A
1    9405
0    1558
Name: count, dtype: int64

Frequency of unique values in 'Label_B':
Label_B
GPT_4-o        1569
gemma-2-9b     1568
mistral-7B     1568
Yi-Large       1568
qwen-2-72B     1567
llama-8B       1565
Human_story    1558
Name: count, dtype: int64

Frequency of unique values in 'best LLM model':
best LLM model
Yi-Large                                       6986
meta-llama/Meta-Llama-3.1-8B-Instruct-Turbo    2763
mistralai/Mistral-7B-Instruct-v0.1              732
Qwen/Qwen2-72B-Instruct                         269
Gemma-3-27b                                     181
Name: count, dtype: int64


In [25]:
# merge the two dataframes on the common columns {'index', 'Text'}
results_GT_df = pd.merge(groundTruth_df, results_df, on=list(common_columns), how='inner')
# save the merged dataframe to a new csv file
results_GT_df.to_csv('results/baseline_benchmark_33_results_merged_with_groundTruth.csv', index=False)

In [26]:
# display(results_GT_df.head())
display(results_GT_df.tail(2))

index                                               Text  Label_A  \
10929  10961  Error : Error communicating with OpenAI : HTTP...        1   
10930  10962  ensemble cast , especially Ayo Edebiri Jeremy ...        1   

       Label_B                               best LLM model  \
10929  GPT_4-o           mistralai/Mistral-7B-Instruct-v0.1   
10930  GPT_4-o  meta-llama/Meta-Llama-3.1-8B-Instruct-Turbo   

       Qwen Qwen2 72B Instruct Edit Score  \
10929                               260.0   
10930                              1978.0   

                        Qwen Qwen2 72B Instruct New Text  \
10929  Regenerated Text: \n\nAn error has occurred: T...   
10930  The ensemble cast, notably featuring Ayo Edebi...   

       mistralai Mistral 7B Instruct v0.1 Edit Score  \
10929                                            2.0   
10930                                         2999.0   

             mistralai Mistral 7B Instruct v0.1 New Text  \
10929   Error: Error communicating with OpenAI: HTTPS...   
10930   The ensemble cast of the show "Eyes Wide Shut...   

       meta llama Meta Llama 3.1 8B Instruct Turbo Edit Score  \
10929                                              389.0        
10930                                             1010.0        

      meta llama Meta Llama 3.1 8B Instruct Turbo New Text  \
10929  Here's the regenerated text with proper format...     
10930  Here's the regenerated text:\n\nThe ensemble c...     

       Yi Large Edit Score                                  Yi Large New Text  \
10929                 18.0  Error: Error communicating with OpenAI: HTTPSC...   
10930               1019.0  Ensemble cast, especially Ayo Edebiri and Jere...   

       Gemma 3 27b Edit Score  \
10929                  4104.0   
10930                  3130.0   

                                    Gemma 3 27b New Text  
10929  Okay, here are a few regenerations of the prov...  
10930  ## Culture Watch: Five Things I’m Engaging Wit...

In [31]:
def replace_values(df, column_name, replacements):
    df[column_name] = df[column_name].replace(replacements)
    return df
# Define the replacements for 'Label_B' and 'best LLM model'
replacements = {
    'gemma-2-9b': 'Gemma',
    'Gemma-3-27b': 'Gemma',
    'mistral-7B': 'Mistral',
    'mistralai/Mistral-7B-Instruct-v0.1': 'Mistral',
    'qwen-2-72B': 'Qwen',
    'Qwen/Qwen2-72B-Instruct': 'Qwen',
    'llama-8B': 'Llama',
    'meta-llama/Meta-Llama-3.1-8B-Instruct-Turbo': 'Llama'
}

# Replace values in 'Label_B' and 'best LLM model' columns
results_GT_df = replace_values(results_GT_df, 'Label_B', replacements)
results_GT_df = replace_values(results_GT_df, 'best LLM model', replacements)

# rename column 'best LLM model' as 'best_LLM_model'
results_GT_df.rename(columns={'best LLM model': 'best_LLM_model'}, inplace=True)

# remove unnecssary columns from the DataFrame and keep only the relevant ones [index	Text	Label_A	Label_B	best LLM model]
results_GT_df = results_GT_df[['index', 'Text', 'Label_A', 'Label_B', 'best_LLM_model']]
# save the modified DataFrame to a new CSV file
results_GT_df.to_csv('results/baseline_benchmark_33_results_merged_with_groundTruth_modified.csv', index=False)

In [32]:
# display the modified DataFrame
display(results_GT_df.tail(2))

index                                               Text  Label_A  \
10929  10961  Error : Error communicating with OpenAI : HTTP...        1   
10930  10962  ensemble cast , especially Ayo Edebiri Jeremy ...        1   

       Label_B best_LLM_model  
10929  GPT_4-o        Mistral  
10930  GPT_4-o          Llama

In [35]:
import pandas as pd
from sklearn.metrics import f1_score, precision_score, recall_score, confusion_matrix, accuracy_score

# Load the dataset
df = pd.read_csv('results/baseline_benchmark_33_results_merged_with_groundTruth_modified.csv')

print("Dataset shape:", df.shape)

# Create a binary classification problem:
# 1 if the predicted model matches the ground truth, 0 otherwise
y_true = [1] * len(df)  # Ideally, all predictions should match ground truth
y_pred = (df['best_LLM_model'] == df['Label_B']).astype(int)

# Calculate metrics
accuracy = accuracy_score(y_true, y_pred)
precision = precision_score(y_true, y_pred)
recall = recall_score(y_true, y_pred)
f1 = f1_score(y_true, y_pred)

# Print results
print(f"Accuracy: {accuracy:.4f}")
print(f"Precision: {precision:.4f}")
print(f"Recall: {recall:.4f}")
print(f"F1 Score: {f1:.4f}")

# Count matches and mismatches
matches = (df['best_LLM_model'] == df['Label_B']).sum()
total = len(df)
print(f"\nMatches: {matches} out of {total} ({matches/total:.4f})")

# For human stories (Label_A == 0)
human_stories = df[df['Label_A'] == 0]
human_matches = (human_stories['best_LLM_model'] == human_stories['Label_B']).sum()
print(f"\nHuman stories (Label_A == 0):")
print(f"Matches: {human_matches} out of {len(human_stories)} ({human_matches/len(human_stories):.4f})")

# For AI-generated content (Label_A == 1)
ai_content = df[df['Label_A'] == 1]
ai_matches = (ai_content['best_LLM_model'] == ai_content['Label_B']).sum()
print(f"\nAI-generated content (Label_A == 1):")
print(f"Matches: {ai_matches} out of {len(ai_content)} ({ai_matches/len(ai_content):.4f})")


Dataset shape: (10931, 5)
Accuracy: 0.1450
Precision: 1.0000
Recall: 0.1450
F1 Score: 0.2533

Matches: 1585 out of 10931 (0.1450)

Human stories (Label_A == 0):
Matches: 0 out of 1547 (0.0000)

AI-generated content (Label_A == 1):
Matches: 1585 out of 9384 (0.1689)


# Results Summary

### Overall Results

• **F1 Score: 0.2533**
• **Accuracy: 14.50%** (1,585 matches out of 10,931 total samples)
• **Precision: 1.0000** (all positive predictions are correct by definition)
• **Recall: 0.1450** (only 14.50% of samples have matching predictions)

### Breakdown by Content Type

1. Human Stories (Label_A == 0):
   • **Matches: 0 out of 1,547 (0.00%)**
   • This makes sense because best_LLM_model doesn't have a "Human_story" option, so it can never 
match with human-written content.

2. AI-Generated Content (Label_A == 1):
   • **Matches: 1,585 out of 9,384 (16.89%)**
   • This means that in only about 17% of cases, the system correctly identified which LLM generated 
the content.